File: frontend/app/api/transactions/hospitalSettings/fetchHospitalSettingsTransaction.ts
import { fetchHospitalSettings } from "../../../api/hospitalSettings/fetchHospitalSettings"
import { normalizeHospitalSettings } from "../../../mapper/hospitalSettingMapper"

type FetchHospitalSettingsTransactionParams = {
    setHospitalSettings: (hospitalSettings: any) => void
}

export async function fetchHospitalSettingsTransaction()
{
    console.log("fetchHospitalSettingsTransaction")

    const hospitalSettings = await fetchHospitalSettings()
    if (!hospitalSettings)
        {return null}
    return normalizeHospitalSettings(hospitalSettings)
}

File: backend/transactions/hospital_settings/fetch_hospital_settings_transaction.py
from hospital_settings.fetch_hospital_settings import fetch_hospital_settings
from supabase import Client

def fetch_hospital_settings_transaction(client:Client,
                                        hospital_id: str):
    print("fetch_hospital_settings_transaction")

    hospital_settings = fetch_hospital_settings(
                                                client,
                                                hospital_id
                                                )

    return hospital_settings